commit bcc55c6664a90146149ba0fd93052adc94287b9f
Author: Steve French <sfrench@us.ibm.com>
Date:   Wed Aug 27 21:30:22 2008 +0000

    [CIFS] Fix plaintext authentication
    
    The last eight bytes of the password field were not cleared when doing lanman plaintext password authentication. This patch fixes that.
    
    I tested it with Samba by setting password
    encryption to no in the server's smb.conf.  Other servers also can be
    configured to force plaintext authentication.    Note that plaintexti
    authentication requires setting /proc/fs/cifs/SecurityFlags to 0x30030
    on the client (enabling both LANMAN and also plaintext password support).
    Also note that LANMAN support (and thus plaintext password support) requires
    CONFIG_CIFS_WEAK_PW_HASH to be enabled in menuconfig.
    
    CC: Jeff Layton <jlayton@redhat.com>
    CC: Stable Kernel <stable@vger.kernel.org>
    Signed-off-by: Steve French <sfrench@us.ibm.com>

diff --git a/fs/cifs/cifsencrypt.c b/fs/cifs/cifsencrypt.c
index 83fd40dc1ef0..bd5f13d38450 100644
--- a/fs/cifs/cifsencrypt.c
+++ b/fs/cifs/cifsencrypt.c
@@ -294,6 +294,7 @@ void calc_lanman_hash(struct cifsSesInfo *ses, char *lnm_session_key)
 
 	if ((ses->server->secMode & SECMODE_PW_ENCRYPT) == 0)
 		if (extended_security & CIFSSEC_MAY_PLNTXT) {
+			memset(lnm_session_key, 0, CIFS_SESS_KEY_SIZE);
 			memcpy(lnm_session_key, password_with_pad,
 				CIFS_ENCPWD_SIZE);
 			return;